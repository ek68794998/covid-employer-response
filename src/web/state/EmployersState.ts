import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}