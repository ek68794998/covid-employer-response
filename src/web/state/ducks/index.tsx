import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});