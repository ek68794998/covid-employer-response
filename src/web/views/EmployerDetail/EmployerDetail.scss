i {
	margin-right: 8px;
}

a {
	text-decoration: none;
}