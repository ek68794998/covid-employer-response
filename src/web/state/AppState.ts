import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}