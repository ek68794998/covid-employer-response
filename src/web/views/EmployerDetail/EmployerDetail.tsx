import React from "react";
import { RouteProps } from "react-router-dom";

import "./EmployerDetail.scss";

interface Props extends RouteProps {
	icon: string;

	iconSize: number;

	link: string;

	text: string;
}

const EmployerDetail: React.FC<Props> = (props: Props): React.ReactElement => {
	const { icon, iconSize, link, text } = props;

	const style: React.CSSProperties = {
		fontSize: iconSize || 24,
	};

	const iconElement: JSX.Element = <i className="EmployerDetail__Icon material-icons" style={style}>{icon}</i>;

	if (link) {
		return (
			<a className="EmployerDetail__Container" href={link} rel="noopener noreferrer" target="_blank">
				{iconElement}
				<span className="EmployerDetail__Label">{text}</span>
			</a>
		);
	}

	return (
		<span className="EmployerDetail__Container">
			{iconElement}
			<span className="EmployerDetail__Label">{text}</span>
		</span>
	);
};

export default EmployerDetail;
