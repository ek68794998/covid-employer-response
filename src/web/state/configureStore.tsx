import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;