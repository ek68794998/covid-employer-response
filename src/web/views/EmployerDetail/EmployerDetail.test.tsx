import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import EmployerDetail from "./EmployerDetail";

describe("<EmployerDetail />", () => {
	test("renders without exploding", () => {
		const store: Store<AppState, AnyAction> = configureStore({});

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<EmployerDetail
							icon={"people"}
							iconSize={36}
							link={"http://example.com"}
							text={"test"}
						/>
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
