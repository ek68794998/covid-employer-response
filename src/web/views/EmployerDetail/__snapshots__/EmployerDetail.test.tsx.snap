// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployerDetail /> renders without exploding 1`] = `
<a
  className="EmployerDetail__Container"
  href="http://example.com"
  rel="noopener noreferrer"
  target="_blank"
>
  <i
    className="EmployerDetail__Icon material-icons"
    style={
      Object {
        "fontSize": 36,
      }
    }
  >
    people
  </i>
  <span
    className="EmployerDetail__Label"
  >
    test
  </span>
</a>
`;
